subsys/bluetooth/host/classic/a2dp.c

```diff
@@ -199,6 +196,10 @@ static int a2dp_set_config_ind(struct bt_avdtp *session, struct bt_avdtp_sep *se
 	__ASSERT(sep, "Invalid sep");
 
 	ep = CONTAINER_OF(sep, struct bt_a2dp_ep, sep);
+	if (ep == NULL) {
```

Comment:
NULL checks for `CONTAINER_OF()` are almost always wrong. It's not a function, rather just pointer arithmetic. The only way you'd get NULL is if you give it a pointer which isn't NULL but has exactly the value `NULL + offsetof(struct bt_a2dp_ep, sep)`. Checking for something like that seems pretty odd to me.

Reply:
Removed the cases and add some more BT_ASSERT for parameter `sep`.